packages/app-builder-lib/src/util/yarn.ts

```diff
@@ -78,13 +79,16 @@ export function getGypEnv(frameworkInfo: DesktopFrameworkInfo, platform: NodeJS.
   }
 }
 
-function checkYarnBerry() {
-  const npmUserAgent = process.env["npm_config_user_agent"] || ""
-  const regex = /yarn\/(\d+)\./gm
+async function checkYarnBerry(pm: PM) {
+  if (pm !== "yarn") {
+    return false
+  }
+  const version = await getNpmVersion(pm)
```

Comment:
Might be worth renaming this function to `getPackageManagerVersion(pm)` as I was confused as to why we were checking for `getNpmVersion` when checking the version of yarn